### PR TITLE
Add PHPStan for Static Analysis

### PR DESCRIPTION
**Pull request by n7studios**
## Summary

[PHPStan](https://phpstan.org/blog/find-bugs-in-your-code-without-writing-tests) performs static analysis on the Plugin's PHP code.  This adds to our tests, PHP Coding Standards and WordPress Coding Standards by ensuring:

- DocBlocks declarations are valid and uniform
- DocBlocks declarations for WordPress `do_action()` and `apply_filters()` calls are valid
- Typehinting variables and return types declared in DocBlocks are correctly cast
- Any unused functions are detected
- Unnecessary checks / code is highlighted for possible removal
- Conditions that do not evaluate can be fixed/removed as necessary

This isn't a replacement for writing tests; rather, an addition to ensure our code continues to meet the high standards expected, reducing ambiguity on how code should be written and ensuring uniformity in what we produce.

## Testing

Instructions are provided in both the [setup](https://github.com/ConvertKit/convertkit-wordpress/blob/phpstan/SETUP.md#phpstan) and [testing](https://github.com/ConvertKit/convertkit-wordpress/blob/phpstan/TESTING.md#run-phpstan) guides for using PHPStan.

The GitHub Action will fail, and a separate PR will follow to ensure code passes static analysis.

## Checklist

* [ ] I have [written a test](TESTING.md#writing-an-acceptance-test) and included it in this PR
* [ ] I have [run all tests](TESTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](TESTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](DEVELOPMENT.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](DEVELOPMENT.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -1,6484 +0,0 @@
-{
```
Comment (noelherrick) [inline composer.lock:1]: Why was this deleted?

**Review by noelherrick — COMMENTED** [review]
I love it. Should we fix the errors then check this in so we don't have failing builds?

File: composer.lock (hunk 1 above)
Comment (n7studios) [reply composer.lock:1]: Don't need to lock dependencies across environments, as all dependencies are used for testing only. Therefore composer should fetch the most up-to-date dependencies to run testing via GitHub Actions and/or locally.

**Comment by n7studios** [comment]
> I love it. Should we fix the errors then check this in so we don't have failing builds?

Nope, because that would go against your request that PRs be smaller in terms of changes for review.  As stated above:
"The GitHub Action will fail, and a separate PR will follow to ensure code passes static analysis."

You're welcome to separately review [that PR here](https://github.com/ConvertKit/convertkit-wordpress/pull/289), once completed and published.

**Review by jenessawhite — APPROVED** [review]
